### PR TITLE
Mmca 4441 - Remove unused code and ToDos through out the services

### PR DESCRIPTION
**Pull request by hari-bytes**

### CODE REVIEW

**Comment by AdamCoops** [comment]
You should be able to resolve some of these platoons errors quite easily.

Config module uk.gov.hmrc.play.audit.AuditModule can be removed from [conf/application.conf](https://github.com/hmrc/customs-financials-email-throttler/blob/MMCA-4441/conf/application.conf#L24). This is already provided by Bootstrap

Config module uk.gov.hmrc.play.bootstrap.graphite.GraphiteMetricsModule can be removed from [conf/application.conf](https://github.com/hmrc/customs-financials-email-throttler/blob/MMCA-4441/conf/application.conf#L28). This is already provided by Bootstrap

Config module uk.gov.hmrc.play.bootstrap.backend.BackendModule can be removed from [conf/application.conf](https://github.com/hmrc/customs-financials-email-throttler/blob/MMCA-4441/conf/application.conf#L31). This is already provided by Bootstrap

File: build.sbt [hunk 1]
```diff
@@ -4,9 +4,17 @@ import uk.gov.hmrc.sbtdistributables.SbtDistributablesPlugin._
 import uk.gov.hmrc.DefaultBuildSettings.targetJvm
```
Comment (tyagikumarvineet) [inline build.sbt]: Could you please address below for this file?

1. Unused import at line 3
2. Please replace _* with * at lines 20 and 21
3. Could you please enable the integration tests? You could refer to customs-data-store changes
    as below
    ThisBuild / majorVersion := 0
ThisBuild / scalaVersion := scala2_13_8

lazy val it = project
  .enablePlugins(PlayScala)
  .dependsOn(microservice % "test->test")
  .settings(itSettings)
  .settings(libraryDependencies ++= Seq("uk.gov.hmrc" %% "bootstrap-test-play-28" % bootstrapVersion % Test))

.configs(IntegrationTest)

File: app/uk/gov/hmrc/customs/financials/emailthrottler/models/EmailRequest.scala [hunk 2]
```diff
@@ -20,7 +20,7 @@ import play.api.libs.functional.syntax._
 import play.api.libs.json.{Format, Json, OFormat}
```
Comment (tyagikumarvineet) [inline app/uk/gov/hmrc/customs/financials/emailthrottler/models/EmailRequest.scala:20]: Could you please split the line at line no 41 as this is very long line. Better to split after =
Once its done, please give a line break after line 42. It should look like as below

  implicit val emailAddressFormat: Format[EmailAddress] =
    implicitly[Format[String]].inmap(EmailAddress, unlift(EmailAddress.unapply))
    
  implicit val emailRequestFormat: OFormat[EmailRequest] = Json.format[EmailRequest]

File: app/uk/gov/hmrc/customs/financials/emailthrottler/services/EmailQueue.scala [hunk 3]
```diff
@@ -76,7 +76,7 @@ class EmailQueue @Inject()(mongoComponent: PlayMongoComponent,
     ).toFutureOption().map {
```
Comment (tyagikumarvineet) [inline app/uk/gov/hmrc/customs/financials/emailthrottler/services/EmailQueue.scala:77]: 1. Could you please give a line break before below lines? 

61, 69, 94,102,108

2. Please remove the unused imports
   import org.mongodb.scala.FindObservable
  import com.mongodb.client.model.ReplaceOptions
 import com.mongodb.client.model.Indexes.descending

File: app/uk/gov/hmrc/customs/financials/emailthrottler/services/MetricsReporterService.scala [hunk 4]
```diff
@@ -84,4 +84,4 @@ class MetricsReporterService @Inject()(val metrics: Metrics, dateTimeService: Da
     metrics.defaultRegistry.counter(counterName).inc()
```
Comment (tyagikumarvineet) [inline app/uk/gov/hmrc/customs/financials/emailthrottler/services/MetricsReporterService.scala:87]: Please split line at line no 51 as its very long line. Please split after =

File: scalastyle-config.xml [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+ <name>Scalastyle standard configuration</name>
+ <check level="warning" class="org.scalastyle.file.FileTabChecker" enabled="true"></check>
+ <check level="warning" class="org.scalastyle.file.FileLengthChecker" enabled="true">
+  <parameters>
+   <parameter name="maxFileLength"><![CDATA[800]]></parameter>
+  </parameters>
+ </check>
+ <check level="warning" class="org.scalastyle.file.HeaderMatchesChecker" enabled="false">
+  <parameters>
+   <parameter name="header"><![CDATA[// Copyright (C) 2011-2012 the original author or authors.
```
Comment (tyagikumarvineet) [inline scalastyle-config.xml]: Could you please align these Lincense related comments properly ? Please refer customs-data-store

File: scalastyle-config.xml [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+ <check level="warning" class="org.scalastyle.scalariform.NoWhitespaceAfterLeftBracketChecker" enabled="true"></check>
+ <check level="warning" class="org.scalastyle.scalariform.ReturnChecker" enabled="true"></check>
+ <check level="warning" class="org.scalastyle.scalariform.NullChecker" enabled="true"></check>
+ <check level="warning" class="org.scalastyle.scalariform.NoCloneChecker" enabled="true"></check>
+ <check level="warning" class="org.scalastyle.scalariform.NoFinalizeChecker" enabled="true"></check>
+ <check level="warning" class="org.scalastyle.scalariform.CovariantEqualsChecker" enabled="true"></check>
+ <check level="warning" class="org.scalastyle.scalariform.StructuralTypeChecker" enabled="true"></check>
+ <check level="warning" class="org.scalastyle.file.RegexChecker" enabled="true">
+  <parameters>
+   <parameter name="regex"><![CDATA[println]]></parameter>
```
Comment (tyagikumarvineet) [inline scalastyle-config.xml:78]: Could you please add print as well as part of additional <parameter>? Please refer customs-data-store

File: scalastyle-config.xml [hunk 7]
```diff
@@ -0,0 +1,117 @@
+<scalastyle>
+ <name>Scalastyle standard configuration</name>
+ <check level="warning" class="org.scalastyle.file.FileTabChecker" enabled="true"></check>
```
Comment (tyagikumarvineet) [inline scalastyle-config.xml]: Could you please update as below as it is becoming a tag with empty body? Same for other places at line 28, 29,30,52, 68-75, 91,92,114-116?

 <check level="warning" class="org.scalastyle.file.FileTabChecker" enabled="true"/>

File: test/test-scalastyle-config.xml [hunk 8]
```diff
@@ -0,0 +1,117 @@
+<scalastyle>
```
Comment (tyagikumarvineet) [inline test/test-scalastyle-config.xml:1]: Same comments (for scalastyle-config.xml) are applicable for this file as well

File: test/uk/gov/hmrc/customs/financials/emailthrottler/controllers/EmailThrottlerControllerSpec.scala [hunk 9]
```diff
@@ -46,7 +46,6 @@ class EmailThrottlerControllerSpec extends SpecBase {
         ),
```
Comment (tyagikumarvineet) [inline test/uk/gov/hmrc/customs/financials/emailthrottler/controllers/EmailThrottlerControllerSpec.scala:46]: Could you please split line no 74 as it is very long line? Better to split after =

File: test/uk/gov/hmrc/customs/financials/emailthrottler/services/EmailJobHandlerSpec.scala [hunk 10]
```diff
@@ -37,11 +37,19 @@ class EmailJobHandlerSpec extends SpecBase {
```
Comment (tyagikumarvineet) [inline test/uk/gov/hmrc/customs/financials/emailthrottler/services/EmailJobHandlerSpec.scala:37]: Could you please address below for this file?
1. 102 and 106 are long lines. PLease split over multiple lines
2. Remove extra line break at line no 129

File: test/uk/gov/hmrc/customs/financials/emailthrottler/services/MetricsReporterServiceSpec.scala [hunk 11]
```diff
@@ -141,9 +141,19 @@ class MetricsReporterServiceSpec extends SpecBase {
   }
```
Comment (tyagikumarvineet) [inline test/uk/gov/hmrc/customs/financials/emailthrottler/services/MetricsReporterServiceSpec.scala:154]: Please address below for this file 

1. Long lines at 155, 156. Once split, please give line break after line 156 and 159. Should look like below
     val startTimestamp: LocalDateTime =
      LocalDateTime.of(tdYear, tdMonth, tdDayOfMonth, tdHour, tdMinute, tdSecondVal30, tdNanoSecond)

    val endTimestamp: LocalDateTime =
      LocalDateTime.of(tdYear, tdMonth, tdDayOfMonth, tdHour, tdMinute, tdSecondVal35, tdNanoSecond)
     2. Remove below comment at line 157
          // endTimestamp - startTimestamp
3. Split long lines at 47, 52, 57, 62
4. Give line breaks before below lines
    75, 87,99,111,123,135

**Comment by tyagikumarvineet** [comment]
Could you please address below as well?

1. graphiteHost in AppConfig seems unused. Please cross check and remove if not used
2. Could you please add a line break each before line 27, 30,33 and 40 in SendEmailJob?
3. For EmailJobHandler, please keep parameters on separate line to reduce the line 24's length as its very long line
    Should look like as below

    class EmailJobHandler @Inject()(emailQueue: EmailQueue,
                                emailNotificationService: EmailNotificationService)(implicit ec: ExecutionContext) {
5. EmailNotificationService - Please give line break before line 37
6. Scheduler - 
   Please split line 35 over multiple lines. Could be done like below
     actorSystem.scheduler
    .scheduleWithFixedDelay(initialDelay = 0 seconds, delay = 1 / appConfig.emailsPerInstancePerSecond second) {
      () => emailJobHandler.processJob()
    }
7. MetricsReporterService - 
    Please give line break before line 52 and 43
     Please split the line 51 as it is very long line. Could be split after = like below

     val elapsedTimeInMillis =
      endTimestamp.toInstant(ZoneOffset.UTC).toEpochMilli - startTimestamp.toInstant(ZoneOffset.UTC).toEpochMilli

**Review by tyagikumarvineet — APPROVED** [review]

**Review by AdamCoops — APPROVED** [review]

**Pull request merged**